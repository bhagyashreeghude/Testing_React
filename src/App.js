// import logo from './logo.svg';
import { useState } from 'react';
import './App.css';

function App() {
  const [ data,setData] =useState("")
  return (
    <div className="App">
      <h1>123456 First React Test case </h1>

      <input type='text' placeholder='user name' name='user-name' id='user-id'/>
      <img  title='nature-img' src='https://cdn.pixabay.com/photo/2015/12/01/20/28/road-1072821_640.jpg' alt='img'></img>
   
      <h1> Testing onChange Event with Input Text</h1>
      {/* <input type='text' value={data} onChange={(e)=>setData(e.target.value)}/> */}
      
      <button onClick={()=>setData("update Data")}>Update data</button>
      <h1>jest testing</h1>
      <h2>hello testing</h2>
      <h1>{data}</h1>
    </div>
  );
}

export default App;
