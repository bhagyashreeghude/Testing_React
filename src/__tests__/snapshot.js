import { render } from "@testing-library/react";
import App from "../App";

test("snapshot testing",()=>{
  const view= render(<App/>);
    expect (view).toMatchSnapshot()
})