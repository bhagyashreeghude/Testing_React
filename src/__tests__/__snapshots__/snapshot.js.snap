// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1>
          123456 First React Test case 
        </h1>
        <input
          id="user-id"
          name="user-name"
          placeholder="user name"
          type="text"
        />
        <img
          alt="img"
          src="https://cdn.pixabay.com/photo/2015/12/01/20/28/road-1072821_640.jpg"
          title="nature-img"
        />
        <h1>
           Testing onChange Event with Input Text
        </h1>
        <button>
          Update data
        </button>
        <h1>
          jest testing
        </h1>
        <h2>
          hello testing
        </h2>
        <h1 />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1>
        123456 First React Test case 
      </h1>
      <input
        id="user-id"
        name="user-name"
        placeholder="user name"
        type="text"
      />
      <img
        alt="img"
        src="https://cdn.pixabay.com/photo/2015/12/01/20/28/road-1072821_640.jpg"
        title="nature-img"
      />
      <h1>
         Testing onChange Event with Input Text
      </h1>
      <button>
        Update data
      </button>
      <h1>
        jest testing
      </h1>
      <h2>
        hello testing
      </h2>
      <h1 />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
